### PR TITLE
fix(types-json): Disable HTML escaping during JSON marshalling

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/plugin-sdk/issues/622

See docs about the default behavior of `json.Marshal`:
https://pkg.go.dev/encoding/json#Marshal
image
See also https://pkg.go.dev/encoding/json#Encoder.SetEscapeHTML:
image

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: schema/json_test.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ func TestJSONSet(t *testing.T) {
 		{source: map[string]Foo{}, result: JSON{Bytes: []byte(`{}`), Status: Present}},
 
 		{source: nil, result: JSON{Status: Null}},
+
+		{source: map[string]any{"test1": "a&b", "test2": "😀"}, result: JSON{Bytes: []byte(`{"test1": "a&b", "test2": "😀"}`), Status: Present}},
```
Comment (hermanschaaf) [inline schema/json_test.go:47]: 😀

Comment (hermanschaaf) [reply schema/json_test.go:47]: I assume this test failed before the fix? 🤔

File: schema/json.go [hunk 2]
```diff
@@ -124,13 +124,17 @@ func (dst *JSON) Set(src any) error {
...
+		encoder.SetEscapeHTML(false)
+		err := encoder.Encode(value)
 		if err != nil {
 			return err
 		}
 
+		buf := buffer.Bytes()
 		// For map and slice jsons, it is easier for users to work with '[]' or '{}' instead of JSON's 'null'.
-		if bytes.Equal(buf, []byte(`null`)) {
+		if bytes.Equal(buf, []byte("null\n")) {
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply schema/json.go:137]: Existing tests are failing without it. I think the encoder handles `nil` values a bit different

File: schema/json_test.go (hunk 1 above)
Comment (erezrokah) [reply schema/json_test.go:47]: Yes, it failed due to the `&` character that was escaped. The emoji is somehow redundant for the purpose of the failure but I wanted to verify it doesn't get escaped too

File: schema/json.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (erezrokah) [reply schema/json.go:137]: Here's why https://github.com/golang/go/blob/5e4e2acc8e53022f57398c4b0f3892f8113687f6/src/encoding/json/stream.go#L220

[the remark this answers is not in the document]
Comment (candiduslynx) [reply schema/json.go:137]: I think that `null` should be preserved as is

File: schema/json_test.go (hunk 1 above)
Comment (candiduslynx) [reply schema/json_test.go:47]: Kindly add the test from https://github.com/cloudquery/cloudquery/pull/7815 as well

Comment (erezrokah) [reply schema/json_test.go:47]: Can you please be specific on which test to add? I think with this fix:
1. Unescaping `{"BillTo":"P\u0026T"}` is not needed at the destination, as the cause was the redundant escaping fixed by this PR
2. `{"BillTo":"P&T"}` should be covered by the test here
3. Not sure why we need to unquote, is there a bug report I can use to reproduce?

Comment (candiduslynx) [reply schema/json_test.go:47]: The unquote one. Basically, it was produced by parsing an AWS output, as the original field was a string representing the JSON contents, and it's returned from the API/SDK escaped/quoted

**Review by candiduslynx — APPROVED** [review]

**Review by hermanschaaf — APPROVED** [review]

**Review by yevgenypats — APPROVED** [review]

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**